synfig-studio/src/gui/dialogs/canvasproperties.cpp

```diff
@@ -56,87 +52,114 @@ using namespace studio;
-	((Gtk::Label *) info_frame->get_label_widget())->set_markup(_("<b>Canvas Info</b>"));
-	dialogBox->pack_start(*info_frame, false, false, 0);
+	connect_signals();
 
-	Gtk::Alignment *infoPadding = manage(new Gtk::Alignment(0, 0, 1, 1));
-	infoPadding->set_padding(6, 0, 24, 0);
-	info_frame->add(*infoPadding);
+	refresh();
+}
 
-	Gtk::Table *info_table=manage(new Gtk::Table(2,2,false));
-	info_table->set_row_spacings(6);
-	info_table->set_col_spacings(12);
-	infoPadding->add(*info_table);
+CanvasProperties::~CanvasProperties()
+{
+}
 
-	// The root canvas doesn't have an ID, so don't
-	// display it if this is a root canvas.
-	if(!canvas_interface_->get_canvas()->is_root())
-	{
-		Gtk::Label *idLabel = manage(new Gtk::Label(_("_ID"), true));
-		idLabel->set_alignment(0, 0.5);
-		idLabel->set_mnemonic_widget(entry_id);
-		info_table->attach(*idLabel, 0, 1, 0, 1, Gtk::SHRINK|Gtk::FILL, Gtk::SHRINK|Gtk::FILL, 0, 0);
-		info_table->attach(entry_id, 1, 2, 0, 1, Gtk::EXPAND|Gtk::FILL, Gtk::SHRINK|Gtk::FILL, 0, 0);
-	}
-	Gtk::Label *nameLabel = manage(new Gtk::Label(_("_Name"), true));
-	nameLabel->set_alignment(0, 0.5);
-	nameLabel->set_mnemonic_widget(entry_name);
-	Gtk::Label *descriptionLabel = manage(new Gtk::Label(_("_Description"), true));
-	descriptionLabel->set_alignment(0, 0.5);
-	descriptionLabel->set_mnemonic_widget(entry_description);
-	info_table->attach(*nameLabel,        0, 1, 1, 2, Gtk::SHRINK|Gtk::FILL, Gtk::SHRINK|Gtk::FILL, 0, 0);
-	info_table->attach(*descriptionLabel, 0, 1, 2, 3, Gtk::SHRINK|Gtk::FILL, Gtk::SHRINK|Gtk::FILL, 0, 0);
-	info_table->attach(entry_name,        1, 2, 1, 2, Gtk::EXPAND|Gtk::FILL, Gtk::SHRINK|Gtk::FILL, 0, 0);
-	info_table->attach(entry_description, 1, 2, 2, 3, Gtk::EXPAND|Gtk::FILL, Gtk::SHRINK|Gtk::FILL, 0, 0);
-
-	dialogBox->pack_start(widget_rend_desc, false, false, 0);
-
-	canvas_interface_->signal_rend_desc_changed().connect(sigc::mem_fun(*this,&studio::CanvasProperties::refresh));
-	canvas_interface_->signal_id_changed().connect(sigc::mem_fun(*this,&studio::CanvasProperties::refresh));
-
-	Gtk::Button *ok_button(manage(new class Gtk::Button(Gtk::StockID("gtk-ok"))));
-	ok_button->show();
-	add_action_widget(*ok_button,2);
-	ok_button->signal_clicked().connect(sigc::mem_fun(*this, &studio::CanvasProperties::on_ok_pressed));
-
-	Gtk::Button *apply_button(manage(new class Gtk::Button(Gtk::StockID("gtk-apply"))));
-	apply_button->show();
-	add_action_widget(*apply_button,1);
-	apply_button->signal_clicked().connect(sigc::mem_fun(*this, &studio::CanvasProperties::on_apply_pressed));
-
-	Gtk::Button *cancel_button(manage(new class Gtk::Button(Gtk::StockID("gtk-close"))));
-	cancel_button->show();
-	add_action_widget(*cancel_button,0);
-	cancel_button->signal_clicked().connect(sigc::mem_fun(*this, &studio::CanvasProperties::on_cancel_pressed));
-
-	//set_default_response(1);
-
-	get_vbox()->show_all();
+void
+CanvasProperties::set_rend_desc(const RendDesc &rend_desc)
+{
+	if (update_lock) return;
```

Comment:
I think it would

I tried removing update_lock previously when I started out working on this PR
Encountered runtime crash
Didn't want to dig deeper so I kept the working code instead

Reply:
Ok but I think this check could be placed anyway as it is:
```
if (rend_desc_ == rend_desc)
	return;
```

Reply:
Testing it. Looks legit and efficient too

Reply:
Semantic issue turns up on IDE
> Invalid operands to binary expression

Still building though. Too many updates so I have 700 files to build... [74/704]

Reply:
Oh. rend_desc doesn't have a comparison operator? :o

Reply:
Build failed
Bunch of errors about synfig::RendDesc not derived from std::pair<...>

The other way is to compare individual property members but that'd be too long

Proceeding with rebase